activerecord/lib/active_record/scoping/default.rb

```diff
@@ -107,7 +107,11 @@ def build_default_scope(base_rel = nil)
 
             if default_scope_override
               # The user has defined their own default scope method, so call that
-              evaluate_default_scope { default_scope }
+              evaluate_default_scope do
+                if scope = default_scope
+                  (base_rel ||= relation).merge(scope)
```

Comment:
👍🏻

Can you think of a way to show the problem (well.. difference, at least) with the other version in a test? I thought I understood what was going on, but couldn't come up with code that proved it.